### DISCUSSION TITLE
Use an entity as reltionship (SIRIUS)

### DISCUSSION THREAD

**Discussion by RaulPiqueras** [eclipse-sirius/sirius-desktop#291] [Q&A] [answered]
Hi everyone, i'm having trouble with my entity in my metamodel in Sirius.

How do I use an entity (of my metamodel) as a relationship in the diagram in Sirius. As you can see in the image, i want to use the entity goal_relationship as a relationship in the diagram. I also attach a image of the result that i want to get.

Thanks!
![Sirius](https://github.com/eclipse-sirius/sirius-desktop/assets/160831422/a4729b3b-cf01-4ade-a875-e3003721f2d9)
![result](https://github.com/eclipse-sirius/sirius-desktop/assets/160831422/ecb10ab3-ea22-432c-ac74-c1c1d1acf4f4)

**Comment by pcdavid** [comment, marked as answer]
Hi, for this case you need to use so-called "Element-Based Edges":

![image](https://github.com/eclipse-sirius/sirius-desktop/assets/10608/a7027f8f-7f9d-4965-b0c1-e53c163c60db)

Like a _Node Mapping_ they use a _Semantic Candidates Expression_ and a _Domain Class_ to identify the model elements they represents (in your case, instances of `Goal_RelationShip`).

They also need a _Source Finder Expression_ and a (set of) _Source Mapping_ (resp. _Target Finder Expression_ and _Target Mapping_) to identify, from the instance (of `Goal_RelationShip`) the edge represents, which nodes are the source (resp. target) to point to.

See [the documentation](https://eclipse.dev/sirius/doc/specifier/diagrams/Diagrams.html#edges) for more details.

You can also have a look at [how the `odesign` of Ecore Tools](https://github.com/eclipse-ecoretools/ecoretools/blob/master/org.eclipse.emf.ecoretools.design/description/ecore.odesign) uses both kinds of edges:
* inheritance relationships (the _EClass.eSuperTypes_ references) are modeled using a _Refrence-Based Edge_ (named _EC ESupertypes_);
* _EReferences_ being themselves actual model elements inside the Ecore (like your `Goal_RelationShip`), are modeled unsing a _Element-Based Edge_ (named _EC_EReference_).

**Comment by RaulPiqueras** [comment]
Hi, How can i create a Element Based Edge? I try it to do it, this it what i do:
![image](https://github.com/eclipse-sirius/sirius-desktop/assets/160831422/81ace22d-dac7-4e19-a588-7cf3a87b9019)

And for create the edge i do it this, but it doesn't work
![image](https://github.com/eclipse-sirius/sirius-desktop/assets/160831422/944df2e5-ce1f-4847-ab90-81620a07500a)

So, how i can do it?
Thanks

**Reply by pcdavid** [reply]
In your screenshot you did not set _Semantic Candidates Expression_ (to tell Sirius which instances of Goal_Relationship to represent with this mapping) nor _Source finder expression_ (from a particular  Goal_Relationship to represent as an edge, which _Goal_ is its source node).
